stumpy/motifs.py

```diff
@@ -0,0 +1,412 @@
+
+    P = P.copy().astype(float)
+    P[P > cutoff] = np.inf
+    P = P[np.newaxis, :]
+
+    result = _motifs(
+        T=T,
+        P=P,
+        M_T=M_T,
+        Σ_T=Σ_T,
+        excl_zone=excl_zone,
+        min_neighbors=min_neighbors,
+        max_distance=max_distance,
+        max_matches=max_matches,
+        max_motifs=max_motifs,
+        normalize=normalize,
+    )
+    result = (
+        _jagged_list_to_array(result[0], fill_value=-1, dtype="int64"),
+        _jagged_list_to_array(result[1], fill_value=np.nan, dtype="float64"),
+    )
+
+    return result
+
+
+def match(
+    Q,
+    T,
+    M_T=None,
+    Σ_T=None,
+    excl_zone=None,
+    max_distance=None,
+    max_matches=None,
+    normalize=True,
+):
+    """
+    Find all matches of a query `Q` in a time series `T`, i.e. the indices
+    of subsequences whose distances to `Q` are less or equal to
+    `max_distance`, sorted by distance (lowest to highest).
+
+    Around each occurrence an exclusion zone is applied before searching for the next.
+
+    Parameters
+    ----------
+    Q : ndarray
+        The query sequence. It doesn't have to be a subsequence of `T`
+
+    T : ndarray
+        The time series of interest
+
+    M_T : ndarray, default None
+        Sliding mean of time series, `T`
+
+    Σ_T : ndarray, default None
+        Sliding standard deviation of time series, `T`
+
+    excl_zone : int, default None
+        Size of the exclusion zone.
+        If `None`, defaults to `m/4`, where `m` is the length of `Q`.
+
+    max_distance : float or function, default None
+        Maximum distance between `Q` and a subsequence `S` for `S` to be considered a
+        match.
+        If a function, then it has to be a function of one argument `D`, which will be
+        the distance profile of `Q` with `T` (a 1D numpy array of size `n-m+1`).
+        If None, defaults to `max(np.mean(D) - 2 * np.std(D), np.min(D))`, i.e. at
+        least the closest match will be returned.
+
+    max_matches : int, default None
+        The maximum amount of similar occurrences to be returned. The resulting
+        occurrences are sorted by distance, so a value of `10` means that the
+        indices of the most similar `10` subsequences is returned. If `None`, then all
+        occurrences are returned.
+
+    normalize : bool, default True
+        When set to `True`, this z-normalizes subsequences prior to computing distances.
+        Otherwise, this function gets re-routed to its complementary non-normalized
+        equivalent set in the `@core.non_normalized` function decorator.
+
+    Returns
+    -------
+    out : ndarray
+        The first column consists of indices of subsequences of `T` whose distances
+        to `Q` are smaller than `atol + rtol * profile_value`, sorted by distance
+        (lowest to highest). The second column consist of the corresponding distances.
+    """
+    if len(Q.shape) == 1:
+        Q = Q[np.newaxis, :]
```

Comment:
You probably want to make a copy here, no? Otherwise, if somebody calls `match` directly then it will change the shape of `Q`

Reply:
No, since `T` is overwritten with a view that has one axis more. You only start having problems (because the new `T` is a view of the old `T`) if you start overwriting elements of `T`.